crates/egui/src/widgets/image.rs

```diff
@@ -288,8 +288,24 @@ impl<'a> Image<'a> {
     }
 
     #[inline]
-    pub fn source(&self) -> &ImageSource<'a> {
-        &self.source
+    pub fn source(&'a self, ctx: &Context) -> ImageSource<'a> {
+        match &self.source {
+            ImageSource::Bytes { uri, .. } => match uri.starts_with("gif://") {
+                true => Some(uri),
+                false => None,
+            },
+            _ => None,
+        }
+        .map(|v| format!("{}-{}", v, get_index(ctx, v)))
```

Comment:
This special format of `{uri}-{frame_index}` should be better documented.

I suggest adding two functions to this file: one for encoding it, one for decoding it (the latter to be used by `gif_loader.rs`).

Reply:
I would also consider using `#` here for the separator. That would be more in line with the `#fragment` part of the URI spec: https://en.wikipedia.org/wiki/Uniform_Resource_Identifier